Team9Project/src/com/indragie/cmput301as1/ExpenseClaimTabFragment.java

```diff
@@ -0,0 +1,482 @@
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.indragie.cmput301as1;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.Response;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.app.ListFragment;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.provider.Settings.Secure;
+import android.util.SparseArray;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.ListView;
+import android.widget.Toast;
+
+
+/**
+ * A fragment base to implement tabs
+ */
+public class ExpenseClaimTabFragment extends ListFragment implements
+TypedObserver<CollectionMutation<ExpenseClaim>> {
+
+	//================================================================================
+	// Constants
+	//================================================================================
+
+	protected static final String EXPENSE_CLAIM_FILENAME = "claims";
+	protected static final String EXPENSE_APPROVAL_FILENAME = "approval";
+	private static final String PREFERENCE = "PREFERENCE";
+	private static final int ADD_EXPENSE_CLAIM_REQUEST = 1;
+	private static final int EDIT_EXPENSE_CLAIM_REQUEST = 2;
+	private static final int SORT_EXPENSE_CLAIM_REQUEST = 3;
+	private static final int MANAGE_TAGS_REQUEST = 4;
+	private static final int FILTER_TAGS_REQUEST = 5;
+
+
+	//================================================================================
+	// Properties
+	//================================================================================
+	/**
+	 *  ListModel of claim items
+	 */
+	protected ListModel<ExpenseClaim> listModel;
+
+	/**
+	 * Index of a item that is long pressed.
+	 */
+	private int longPressedItemIndex;
+
+	/**
+	 * Active user.
+	 */
+	protected User user;
+
+	/** Activity of the class that created the fragment */
+
+	protected Activity activity;
+
+	/**
+	 * Manages the user and associated preferences.
+	 */
+	protected UserManager userManager;
+	
+	/**
+	 * List of tags to filter expense claims.
+	 */
+	private ArrayList<Tag> filteredTagsList = new ArrayList<Tag>();
+	
+	/**
+	 * List Model of filtered expense claim.
+	 */
+	protected ListModel<ExpenseClaim> filteredListModel;
+
+	//================================================================================
+	// Activity Callbacks
+	//================================================================================
+
+	
+	
+	@Override
+	public void onActivityCreated(Bundle savedInstanceState) {
+		super.onActivityCreated(savedInstanceState);
+
+
+		getListView().setOnItemLongClickListener(
+				new LongClickDeleteListener(activity, new LongClickDeleteListener.OnDeleteListener() {
```

Comment:
Looks like this part wasn't merged correctly? The implementation in master is different.